pandas/core/series.py

```diff
@@ -2558,8 +2571,12 @@ def _reindex_indexer(self, new_index, indexer, copy):
                 return self.copy()
             return self
 
-        # be subclass-friendly
-        new_values = algorithms.take_1d(self.get_values(), indexer)
+        if is_sparse(self):
```

Comment:
huh? this should be handled in take (I think it already is?)

Reply:
Nope (https://github.com/pandas-dev/pandas/issues/19506)

```python
In [5]: arr = pd.SparseArray([1, 0, 1, 2], fill_value=0)

In [6]: pd.core.algorithms.take_1d(arr, [0, 1])
Out[6]: array([1, 1])

In [7]: pd.core.algorithms.take_1d(arr.get_values(), [0, 1])
Out[7]: array([1, 0])
```

Reply:
no my point is it *should* be handled there. I would simply move it there now (where you can do the is_sparse check). The point is the ``.get_values()`` should be there not in this part of the code.

Reply:
@TomAugspurger If you pass `_values` also for sparse, you can still do `get_values()` in `take_nd`, as this is also defined on the sparse array.